### ISSUE TITLE
Prisma Client is unable to fetch DateTime lists

### ISSUE THREAD

**Issue by madebysid** [prisma/prisma#3517]
## Bug description

Prisma Client is unable to fetch DateTime lists

## How to reproduce

Use this schema:
```
generator client {
  provider = "prisma-client-js"
}

datasource ds {
  provider = "postgresql"
  url      = "postgresql://siddhant@localhost:5432/experiments" // Replace
}

model User {
  id Int @id

  dateTimeList DateTime[]
}
```

& use this script:

```
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();

const main = async () => {
  await prisma.user.create({
    data: {
      id: 1,
      dateTimeList: {
        set: [
          new Date().toISOString(),
          new Date().toISOString(),
          new Date().toISOString(),
        ],
      },
    },
  });

  console.log(
    await prisma.user.findMany({
      where: {
        id: 1,
      },
    })
  );
};

main()
  .catch(async (e) => console.log(e))
  .finally(async () => {
    await prisma.$disconnect();
  });

```

The output of this script is:

```
[ { id: 1, dateTimeList: Invalid Date } ]
```

## Expected behavior

Expected output:

```
[ { id: 1, dateTimeList: [

   "2020-09-03T20:28:05.886Z"
   "2020-09-03T20:28:05.886Z",
   "2020-09-03T20:28:05.886Z"
] } ]
```

Actual data in the DB seems to be correct AFAICT:
![Screenshot 2020-09-03 at 22 30 41](https://user-images.githubusercontent.com/7689783/92164136-259dd700-ee35-11ea-8f2a-b393c6eae994.png)

## Environment & setup

- OS: macOS
- Database: PostgreSQL
- Node.js version: v12.16.3
- Prisma version:
```
@prisma/cli          : 2.6.1
Current platform     : darwin
Query Engine         : query-engine 6a8054bb549e4cc23f157b0010cb2e95cb2637fb (at node_modules/@prisma/cli/query-engine-darwin)
Migration Engine     : migration-engine-cli 6a8054bb549e4cc23f157b0010cb2e95cb2637fb (at node_modules/@prisma/cli/migration-engine-darwin)
Introspection Engine : introspection-core 6a8054bb549e4cc23f157b0010cb2e95cb2637fb (at node_modules/@prisma/cli/introspection-engine-darwin)
Format Binary        : prisma-fmt 6a8054bb549e4cc23f157b0010cb2e95cb2637fb (at node_modules/@prisma/cli/prisma-fmt-darwin)
Studio               : 0.272.0
```

**Comment by pantharshit00** [comment]
I can confirm this, thanks for the detailed reproduction as always.

**Comment by janpio** [comment]
@pantharshit00 Please look into if this is TS or Engines by looking at the query generated

**Comment by pantharshit00** [comment]
Error happens at TS level during date conversion. Engine is returning correct data:
![image](https://user-images.githubusercontent.com/22195362/96241625-111a3600-0fc0-11eb-9701-a4d832c86809.png)

**Comment by pantharshit00** [comment]
cc @Jolg42 @timsuchanek

**Comment by timsuchanek** [comment]
Just tried to reproduce, but can't.
It's very possible, that we fixed this in `2.10.0`, because we touched related logic.
@pantharshit00 and @madebysid  can you try to reproduce again with the latest dev (`2.11.0-dev.14`) ?

**Comment by madebysid** [comment]
Just tried this, and it seems to be working as expected now, closing!
